java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -164,6 +167,14 @@ private void readVector(Field field, FieldVector vector) throws JsonParseExcepti
     readToken(END_OBJECT);
   }
 
+  private byte[] decodeHexSafe(String hexString) throws IOException {
+    try {
+      return Hex.decodeHex(hexString.toCharArray());
+    } catch (DecoderException e) {
+      throw new IOException("Unable to decode hex string: " + hexString);
```

Comment:
Exceptions should be chained:
throw new IOException("Unable to decode hex string: " + hexString, e);